modin/core/execution/dask/implementations/pandas_on_dask/dataframe/dataframe.py

```diff
@@ -41,22 +42,63 @@ class PandasOnDaskDataframe(PandasDataframe):
 
     _partition_mgr_cls = PandasOnDaskDataframePartitionManager
 
+    def _get_partition_size_along_axis(self, partition, axis=0):
```

Comment:
Ah, I'm wrong about this function blocking on inner partitions, because it doesn't call `length` or `width`. So I think it's correct. Still, this part seems outside the scope of this PR. I'd rather worry about this fix in a separate PR for #4494.

Reply:
I think it makes sense to include here, since the original code to get length + width in parallel breaks when applied to axis partitions so I'll need to fix that code anyways, and if I'm doing that, I may as well fix the code to just do it all in parallel right?

Reply:
If the original code was broken, I think it's okay to include the partial fix for #4494 here.